libutils/string_lib.c

```diff
@@ -59,15 +59,13 @@ char *StringFormat(const char *fmt, ...)
     return res;
 }
 
-unsigned int StringHash(const char *str, unsigned int seed, unsigned int max)
+unsigned int StringHash(const char *str, unsigned int seed)
 {
-    CF_ASSERT(ISPOW2(max),
```

Comment:
Shouldn't this assert still be somewhere, since you are still using binary and instead of division ?

Reply:
Not really. The *HashMap* now makes sure its size is a power of 2 and it's the hash map doing the limitation of the hash values returned by hash functions. See https://github.com/cfengine/core/pull/2915/files#diff-557504754a8a166d13cf3066e34c1390R49 for details.

Reply:
Then it should be inside GetBucket, right before the binary and computation. I agree that it's an expensive assertion to have there though. Maybe use `assert()` which is compiled out in release builds.